src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -63,29 +72,54 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         /// <param name="token">The JWT token.</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
+            bool result;
+
             try
             {
-                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
-                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
+                var validationParameters = await DetermineTokenValidationParametersAsync();
+                _handler.ValidateToken(token, validationParameters, out var validatedToken);
+                result = ValidateClaimCheck(validatedToken);
             }
-            catch (Exception exception)
+            catch
             {
                 return false;
             }
 
-            return true;
+            return result;
+        }
+
+        /// <summary>
+        /// Verifies if the claims coming in ClaimCheck object vs the claims in the jwt object based on the claim name and value are valid
+        /// </summary>
+        /// <param name="validatedToken">Security token which contains the jwt claims</param>
+        /// <returns></returns>
+        public bool ValidateClaimCheck(SecurityToken validatedToken)
+        {
+            if (_claimCheck == null || !_claimCheck.Any())
+            {
+                return true;
+            }
+
+            var jwtToken = (JwtSecurityToken) validatedToken;
+
+            var validClaims = (from c in jwtToken.Claims
+                join m in _claimCheck on c.Type equals m.Key
+                where c.Value == m.Value
+                select c).Distinct().Count();
+
+            return _claimCheck.Count == validClaims;
         }
 
         private async Task<TokenValidationParameters> DetermineTokenValidationParametersAsync()
         {
             if (_tokenValidationParameters is null)
             {
-                OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
-                
+                var config = await _configManager.GetConfigurationAsync();
```

Comment:
I think we tend to specify the type explicitly if we assign a method return value to it.
I'm sure @stijnmoreels can tell me if I'm right or wrong on this one

Reply:
Yes, that right! The Arcus code guidelines specify that we should specify the type instead of using `var` if we don't know that the type is. So, in this case, it should be `OpenIdConnectionConfiguration`.

Reply:
done